pandas/core/series.py

```diff
@@ -5642,18 +5642,15 @@ def to_timestamp(
         -------
         Series with DatetimeIndex
         """
-        new_values = self._values
-        if copy:
-            new_values = new_values.copy()
+        new_obj = self.copy(deep=copy)
```

Comment:
Do this copy after the index isinstance check?

Reply:
Moved, kept the df as is to keep it consistent with the previous behavior